mobileApp/src/machines/gameMachine/gameMachine.ts

```diff
@@ -0,0 +1,112 @@
+import {assign, Machine} from "xstate";
+
+import {IInitState} from "../../redux/reducers";
+import {calculateWinner} from "../../utils/calculateWinner";
+
+const initialState: IInitState = {
+    history: [
+        {
+            squares: Array(9).fill(null),
+        },
+    ],
+    xIsNext: true,
+    stepNumber: 0,
+    winner: null,
+    disabled: false,
+};
+
+interface GameMachineStateSchema {
+    states: {
+        idle: {},
+        xTurn: {},
+        oTurn: {},
+    },
+}
+
+export const gameMachine = Machine<IInitState, GameMachineStateSchema>({
+        id: 'gameMachine',
+        initial: 'idle',
+        context: initialState,
+        states: {
+            idle: {
+                on: {
+                    TRANSITION: {
+                        target: 'xTurn',
+                        actions:
+                            assign((context, event, payload) => {
```

Comment:
Please, make a function for this assignment and the assignment below and use it for both. They are almost the same.